plugin/src/main/java/org/opensearch/ml/model/MLModelAutoReLoader.java

```diff
@@ -0,0 +1,271 @@
+     * @return SearchResponse
+     */
+    @VisibleForTesting
+    SearchResponse queryTask(String nodeId) {
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder().from(0).size(1);
+
+        QueryBuilder queryBuilder = QueryBuilders
+            .boolQuery()
+            .must(QueryBuilders.matchPhraseQuery("task_type", "LOAD_MODEL"))
+            .must(QueryBuilders.matchPhraseQuery("state", "COMPLETED"))
+            .must(QueryBuilders.matchPhraseQuery("worker_node", nodeId));
+        searchSourceBuilder.query(queryBuilder);
+
+        SortBuilder sortBuilderOrder = new FieldSortBuilder("create_time").order(SortOrder.DESC);
+        searchSourceBuilder.sort(sortBuilderOrder);
+
+        SearchRequest searchRequest = new SearchRequest().source(searchSourceBuilder).indices(ML_TASK_INDEX);
+        try {
+            return client.execute(SearchAction.INSTANCE, searchRequest).actionGet(5000);
+        } catch (IndexNotFoundException e) {
+            log.error("index {} not found, the reason is {}", ML_TASK_INDEX, e.getMessage());
+            throw new IndexNotFoundException("index " + ML_TASK_INDEX + " not found");
+        }
+    }
+
+    /**
+     *  auto reload 1 model under the node id
+     * @param nodeId node id
+     * @param modelId model id
+     */
+    @VisibleForTesting
+    LoadModelResponse autoReLoadModelByNodeAndModelId(String nodeId, String modelId) {
+        MLLoadModelRequest mlLoadModelRequest = new MLLoadModelRequest(modelId, new String[] { nodeId }, false, false);
+        return client.execute(MLLoadModelAction.INSTANCE, mlLoadModelRequest).actionGet(5000);
+    }
+
+    /**
+     * get retry times from the index ".plugins-ml-model-reload" by 1 ml node
+     * @param nodeId the filter condition to query
+     * @return retry times
+     */
+    @VisibleForTesting
+    int getReTryTimes(String nodeId) {
+        if (!isExistedIndex(ML_MODEL_RELOAD_INDEX)) {
+            return 0;
+        }
+
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+        searchSourceBuilder.fetchSource(new String[] { MODEL_LOAD_RETRY_TIMES_FIELD }, null);
+        QueryBuilder queryBuilder = QueryBuilders.idsQuery().addIds(nodeId);
+        searchSourceBuilder.query(queryBuilder);
+        SearchRequest searchRequest = new SearchRequest().source(searchSourceBuilder).indices(ML_MODEL_RELOAD_INDEX);
+
+        try {
+            SearchResponse response = client.execute(SearchAction.INSTANCE, searchRequest).actionGet(5000);
+
+            SearchHit[] hits = response.getHits().getHits();
+            if (CollectionUtils.isEmpty(hits)) {
+                return 0;
+            }
+
+            Map<String, Object> sourceAsMap = hits[0].getSourceAsMap();
+            return (Integer) sourceAsMap.get(MODEL_LOAD_RETRY_TIMES_FIELD);
+        } catch (IndexNotFoundException e) {
+            throw new IndexNotFoundException("index " + ML_MODEL_RELOAD_INDEX + " not found");
```

Comment:
add log here.

Reply:
it we throw the exception, it is no need to log again

Reply:
Suddenly, I think it no need to catch `IndexNotFoundException`. Because I have checked index whether existed at the begin of this method when the method is invoked at the first time. So in the part of query executed, the index must have existed~